robomaker-sample-app-ci/src/aws-robomaker-sample-application-ci.ts

```diff
@@ -193,13 +199,25 @@ async function build() {
 }
 
 async function bundle() {
-  try {
-    const bundleFilename = path.basename(WORKSPACE_DIRECTORY); 
-    await exec.exec("colcon", ["bundle", "--build-base", "build", "--install-base", "install", "--bundle-base", "bundle"], getWorkingDirExecOptions());
-    await exec.exec("mv", ["bundle/output.tar", `../${bundleFilename}.tar`], getWorkingDirExecOptions());
-    await exec.exec("rm", ["-rf", "bundle"], getWorkingDirExecOptions());  // github actions have been failing with no disk space
-  } catch (error) {
-    core.setFailed(error.message);
+  let delay_ms = 1000 * MINIMUM_BACKOFF_TIME_SECONDS; 
+  // indexed from 0 because COLCON_BUNDLE_RETRIES is the number of retries AFTER the initial try
+  for (let i = 0; i <= COLCON_BUNDLE_RETRIES; i++) {
+    try {
+      const bundleFilename = path.basename(WORKSPACE_DIRECTORY);
+      await exec.exec("colcon", ["bundle", "--build-base", "build", "--install-base", "install", "--bundle-base", "bundle"], getWorkingDirExecOptions());
+      await exec.exec("mv", ["bundle/output.tar", `../${bundleFilename}.tar`], getWorkingDirExecOptions());
+      await exec.exec("rm", ["-rf", "bundle"], getWorkingDirExecOptions());  // github actions have been failing with no disk space
+      break; // break if colcon bundle passes
+    } catch (error) {
+      await exec.exec("rm", ["-rf", "bundle"], getWorkingDirExecOptions()); // remove erred bundle assets
```

Comment:
Why remove the bundle?

Reply:
Remove the bundle only in case of colcon bundle failure.. Not sure if it is recommended but the idea behind removing the bundle was to remove any (stale) cached metadata if any which might cause hash sum mismatch during the retry.. I can double-check what's the recommendation from `colcon bundle`.. if the cached metadata is not an issue, the cached data might speed up the retry..

Reply:
Remove the bundle only in case of colcon bundle failure.. Not sure if it is recommended but the idea behind removing the bundle was to remove any (stale) cached metadata if any which might cause hash sum mismatch during the retry.. I can double-check what's the recommendation from `colcon bundle`..

Reply:
I am not entirely confident that the dirty retry won't cause the hashsum mismatch (the problem we are trying to run away from).. if it does indeed cause hashsum mismatch due to a stale cache(sources list, metadata etc.), our retries are basically redundant.. a clean retry is a better option IMO even though it's a bit time consuming..